Work-around for:
https://github.com/emscripten-core/emscripten/issues/6759

Probably here to stay until emterpreter is re-implemented on top of
the new Emscripten LLVM back-end, which doesn't need
EMULATED_FUNCTION_POINTER_CASTS

Index: renpy/renpy/gl/gldraw.pyx
===================================================================
--- renpy.orig/renpy/gl/gldraw.pyx
+++ renpy/renpy/gl/gldraw.pyx
@@ -1492,7 +1492,7 @@ cdef class Environ(object):
         Sets the array of texture coordinates for unit `unit`.
         """
 
-    cdef void set_color(self, float r, float g, float b, float a):
+    cdef void set_color(self, double r, double g, double b, double a):
         """
         Sets the color to be shown.
         """
Index: renpy/renpy/gl/gl.pxd
===================================================================
--- renpy.orig/renpy/gl/gl.pxd
+++ renpy/renpy/gl/gl.pxd
@@ -1468,10 +1468,10 @@ cdef extern from "glcompat.h" nogil:
     GLvoid realGlLinkProgramARB "glLinkProgramARB" (GLhandleARB)
     GLvoid realGlUseProgramObjectARB "glUseProgramObjectARB" (GLhandleARB)
     GLvoid realGlValidateProgramARB "glValidateProgramARB" (GLhandleARB)
-    GLvoid realGlUniform1fARB "glUniform1fARB" (GLint, GLfloat)
-    GLvoid realGlUniform2fARB "glUniform2fARB" (GLint, GLfloat, GLfloat)
-    GLvoid realGlUniform3fARB "glUniform3fARB" (GLint, GLfloat, GLfloat, GLfloat)
-    GLvoid realGlUniform4fARB "glUniform4fARB" (GLint, GLfloat, GLfloat, GLfloat, GLfloat)
+    GLvoid realGlUniform1fARB "glUniform1fARB" (GLint, double)
+    GLvoid realGlUniform2fARB "glUniform2fARB" (GLint, double, double)
+    GLvoid realGlUniform3fARB "glUniform3fARB" (GLint, double, double, double)
+    GLvoid realGlUniform4fARB "glUniform4fARB" (GLint, double, double, double, double)
     GLvoid realGlUniform1iARB "glUniform1iARB" (GLint, GLint)
     GLvoid realGlUniform2iARB "glUniform2iARB" (GLint, GLint, GLint)
     GLvoid realGlUniform3iARB "glUniform3iARB" (GLint, GLint, GLint, GLint)
@@ -1562,10 +1562,10 @@ cdef int glShaderSourceARB(GLhandleARB,
 cdef int glTexImage2D(GLenum, GLint, GLint, GLsizei, GLsizei, GLint, GLenum, GLenum, GLubyte *) except? 0
 cdef int glTexParameteri(GLenum, GLenum, GLint) except? 0
 cdef int glTexSubImage2D(GLenum, GLint, GLint, GLint, GLsizei, GLsizei, GLenum, GLenum, GLubyte *) except? 0
-cdef int glUniform1fARB(GLint, GLfloat) except? 0
+cdef int glUniform1fARB(GLint, double) except? 0
 cdef int glUniform1iARB(GLint, GLint) except? 0
-cdef int glUniform2fARB(GLint, GLfloat, GLfloat) except? 0
-cdef int glUniform4fARB(GLint, GLfloat, GLfloat, GLfloat, GLfloat) except? 0
+cdef int glUniform2fARB(GLint, double, double) except? 0
+cdef int glUniform4fARB(GLint, double, double, double, double) except? 0
 cdef int glUniformMatrix4fvARB(GLint, GLsizei, GLboolean, GLfloat *) except? 0
 cdef int glUseProgramObjectARB(GLhandleARB) except? 0
 cdef int glVertexAttribPointerARB(GLuint, GLint, GLenum, GLboolean, GLsizei, GLubyte *) except? 0
Index: renpy/renpy/gl/gl.pyx
===================================================================
--- renpy.orig/renpy/gl/gl.pyx
+++ renpy/renpy/gl/gl.pyx
@@ -593,7 +593,7 @@ cdef int glTexSubImage2D(GLenum a0, GLin
                 raise Exception(message)
     return 1
 
-cdef int glUniform1fARB(GLint a0, GLfloat a1) except? 0:
+cdef int glUniform1fARB(GLint a0, double a1) except? 0:
     if check_errors & 4:
         renpy.display.log.write('glUniform1fARB')
     cdef GLenum error
@@ -623,7 +623,7 @@ cdef int glUniform1iARB(GLint a0, GLint
                 raise Exception(message)
     return 1
 
-cdef int glUniform2fARB(GLint a0, GLfloat a1, GLfloat a2) except? 0:
+cdef int glUniform2fARB(GLint a0, double a1, double a2) except? 0:
     if check_errors & 4:
         renpy.display.log.write('glUniform2fARB')
     cdef GLenum error
@@ -638,7 +638,7 @@ cdef int glUniform2fARB(GLint a0, GLfloa
                 raise Exception(message)
     return 1
 
-cdef int glUniform4fARB(GLint a0, GLfloat a1, GLfloat a2, GLfloat a3, GLfloat a4) except? 0:
+cdef int glUniform4fARB(GLint a0, double a1, double a2, double a3, double a4) except? 0:
     if check_errors & 4:
         renpy.display.log.write('glUniform4fARB')
     cdef GLenum error
Index: renpy/renpy/gl/gldraw.pxd
===================================================================
--- renpy.orig/renpy/gl/gldraw.pxd
+++ renpy/renpy/gl/gldraw.pxd
@@ -89,7 +89,7 @@ cdef class Environ:
     cdef void imageblend(self, double fraction, int ramp)
     cdef void set_vertex(self, float *vertices)
     cdef void set_texture(self, int unit, float *coords)
-    cdef void set_color(self, float r, float g, float b, float a)
+    cdef void set_color(self, double r, double g, double b, double a)
     cdef void set_clip(self, tuple clip_box, GLDraw draw)
     cdef void unset_clip(self, GLDraw draw)
     cdef void ortho(self, double left, double right, double bottom, double top, double near, double far)
Index: renpy/renpy/gl/glenviron_fixed.pyx
===================================================================
--- renpy.orig/renpy/gl/glenviron_fixed.pyx
+++ renpy/renpy/gl/glenviron_fixed.pyx
@@ -319,7 +319,7 @@ cdef class FixedFunctionEnviron(Environ)
         else:
             glDisableClientState(GL_TEXTURE_COORD_ARRAY)
 
-    cdef void set_color(self, float r, float g, float b, float a):
+    cdef void set_color(self, double r, double g, double b, double a):
         glColor4f(r, g, b, a)
 
     cdef void ortho(self, double left, double right, double bottom, double top, double near, double far):
Index: renpy/renpy/gl/glenviron_limited.pyx
===================================================================
--- renpy.orig/renpy/gl/glenviron_limited.pyx
+++ renpy/renpy/gl/glenviron_limited.pyx
@@ -182,7 +182,7 @@ cdef class LimitedEnviron(Environ):
         else:
             glDisableClientState(GL_TEXTURE_COORD_ARRAY)
 
-    cdef void set_color(self, float r, float g, float b, float a):
+    cdef void set_color(self, double r, double g, double b, double a):
         glColor4f(r, g, b, a)
 
 
Index: renpy/renpy/gl/glenviron_shader.pyx
===================================================================
--- renpy.orig/renpy/gl/glenviron_shader.pyx
+++ renpy/renpy/gl/glenviron_shader.pyx
@@ -566,7 +566,7 @@ cdef class ShaderEnviron(Environ):
         else:
             glDisableVertexAttribArrayARB(tex)
 
-    cdef void set_color(self, float r, float g, float b, float a):
+    cdef void set_color(self, double r, double g, double b, double a):
         glUniform4fARB(self.program.Color, r, g, b, a)
 
     cdef void ortho(self, double left, double right, double bottom, double top, double near, double far):
